src/networking/APNSNetworkService.php

```diff
@@ -0,0 +1,140 @@
+<?php
+declare( strict_types = 1 );
+
+class APNSNetworkService {
+
+	/** @var resource **/
+	private $curl_handle;
+
+	/** @var int */
+	public $port = 443;
+
+	/** @var bool **/
+	private $debug = false;
+
+	/** @var bool **/
+	private $ssl_verification_enabled = true;
+
+	public function __construct() {
+		$ch = curl_multi_init();
+		curl_multi_setopt( $ch, CURLMOPT_PIPELINING, CURLPIPE_MULTIPLEX );
+		curl_multi_setopt( $ch, CURLMOPT_MAX_TOTAL_CONNECTIONS, 1 );
+		// TODO: Setting CURLOPT_PIPEWAIT results in the following error
+		//
+		// ```
+		// PHP Warning:  curl_multi_setopt(): Invalid curl multi configuration option in php-push/src/networking/APNSNetworkService.php on line 19
```

Comment:
The message says line 19, because that's where the call was at the time, even though it would be at line 22 now.

I wonder if it's worth redacting the line, or whether it doesn't matter because future us are going to try the code and get a fresh stacktrace anyways.